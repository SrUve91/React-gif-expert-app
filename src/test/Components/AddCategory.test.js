import React from 'react'
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories =  { setCategories }/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories =  { setCategories }/>);
    });

    test('Debe coincidir con el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change',{ target:{ value: value} });

        const inputAfter = wrapper.find('input');
        expect( inputAfter.props().value).toBe(value);

    });    

    test('No debe de hacer post de la informacion OnSubmit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change',{ target:{ value: value} });
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        const inputAfter = wrapper.find('input');

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect( inputAfter.props().value).toBe('');
    })
    
    
});
