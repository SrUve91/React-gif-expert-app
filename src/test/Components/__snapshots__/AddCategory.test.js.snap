// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe coincidir con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
